From bca7975b39efea16e6e669853fe897d0d6bd89f7 Mon Sep 17 00:00:00 2001
From: "commit-queue@webkit.org"
 <commit-queue@webkit.org@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date: Fri, 3 Sep 2021 12:07:18 +0000
Subject: [PATCH] Silence more -Wreturn-type warnings
 https://bugs.webkit.org/show_bug.cgi?id=229858

Unreviewed.

Patch by Michael Catanzaro <mcatanzaro@gnome.org> on 2021-09-03

* runtime/TemporalDuration.cpp:
(JSC::propertyName):
* runtime/TemporalObject.cpp:
(JSC::roundNumberToIncrement):

git-svn-id: http://svn.webkit.org/repository/webkit/trunk@281986 268f45cc-cd09-0410-ab3c-d52691b4dbfc
---
 Source/JavaScriptCore/ChangeLog                    | 12 ++++++++++++
 Source/JavaScriptCore/runtime/TemporalDuration.cpp |  2 ++
 Source/JavaScriptCore/runtime/TemporalObject.cpp   |  2 ++
 3 files changed, 16 insertions(+)

diff --git a/Source/JavaScriptCore/runtime/TemporalDuration.cpp b/Source/JavaScriptCore/runtime/TemporalDuration.cpp
index 67ad4e819ff4..0da24a52557e 100644
--- a/Source/JavaScriptCore/runtime/TemporalDuration.cpp
+++ b/Source/JavaScriptCore/runtime/TemporalDuration.cpp
@@ -42,6 +42,8 @@ static PropertyName propertyName(VM& vm, unsigned index)
         JSC_TEMPORAL_UNITS(JSC_TEMPORAL_DURATION_PROPERTY_NAME)
 #undef JSC_TEMPORAL_DURATION_PROPERTY_NAME
     }
+
+    RELEASE_ASSERT_NOT_REACHED();
 }
 
 const ClassInfo TemporalDuration::s_info = { "Object", &Base::s_info, nullptr, nullptr, CREATE_METHOD_TABLE(TemporalDuration) };
diff --git a/Source/JavaScriptCore/runtime/TemporalObject.cpp b/Source/JavaScriptCore/runtime/TemporalObject.cpp
index 9535719f7ec1..e9b872d2418a 100644
--- a/Source/JavaScriptCore/runtime/TemporalObject.cpp
+++ b/Source/JavaScriptCore/runtime/TemporalObject.cpp
@@ -328,6 +328,8 @@ double roundNumberToIncrement(double x, double increment, RoundingMode mode)
     case RoundingMode::HalfExpand:
         return std::round(quotient) * increment;
     }
+
+    RELEASE_ASSERT_NOT_REACHED();
 }
 
 } // namespace JSC
-- 
2.32.0

